lib/Parse/ParseExpr.cpp

```diff
@@ -2288,8 +2288,13 @@ Expr *Parser::parseExprIdentifier() {
     } else {
       for (auto activeVar : DisabledVars) {
         if (activeVar->getFullName() == name) {
-          diagnose(loc.getBaseNameLoc(), DisabledVarReason);
-          return new (Context) ErrorExpr(loc.getSourceRange());
+          if (DisabledVarReason.ID == diag::var_init_self_referential.ID) {
```

Comment:
How about not disabling the var at all in this case then?

Reply:
Hmm yeah that sounds like a better idea than this if check!